pipeline/e2e-job-per-environment.yaml

```diff
@@ -3,30 +3,63 @@
 
 parameters:
     jobNameSuffix: ''
-    windowsImage: 'windows-2019'
+    windowsImage: 'windows-2022-secure'
     macImage: 'macOS-11'
-    linuxImage: 'ubuntu-20.04'
+    linuxImage: 'ubuntu-22.04-secure'
 
 jobs:
     - job: 'e2e_tests_windows${{ parameters.jobNameSuffix }}'
+      templateContext:
+          outputs:
+              - output: pipelineArtifact
+                targetPath: '$(System.DefaultWorkingDirectory)/test-results/e2e'
+                artifactName: '$(Agent.JobName)-screenshots-and-extended-logs'
+                condition: failed() # because the detailed chrome logs are ~150MB and uploading them takes ~30s
```

Comment:
This comment was (and still is) somewhat cryptic. We only publish the e2e screenshots and logs if the e2e tests failed. We do this because publishing the information is slow and takes storage space. If there are no errors, we don't want to pay this penalty

Reply:
we set condition to failed so it will output only in case of failures